File: src/main/java/com/gg/plugins/mongo/model/MongoCollection.java
/*
 * Copyright (c) 2022. Geetesh Gupta
 */

package com.gg.plugins.mongo.model;

import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public class MongoCollection implements Comparable<MongoCollection> {

	private final String name;

	private final MongoDatabase parentDatabase;

	public MongoCollection(String name, MongoDatabase mongoDatabase) {
		this.name = name;
		this.parentDatabase = mongoDatabase;
	}

	public MongoDatabase getParentDatabase() {
		return parentDatabase;
	}

	@Override
	public int compareTo(@NotNull MongoCollection otherCollection) {
		return this.name.compareTo(otherCollection.getName());
	}

	public String getName() {
		return name;
	}

	@Override
	public int hashCode() {
		return Objects.hash(name, parentDatabase);
	}

	@Override
	public boolean equals(Object o) {
		if (this == o)
			return true;
		if (!(o instanceof MongoCollection))
			return false;
		MongoCollection that = (MongoCollection) o;
		return Objects.equals(name, that.name) && Objects.equals(parentDatabase, that.parentDatabase);
	}

	@Override
	public String toString() {
		return "MongoCollection{" + "name='" + name + '\'' + '}';
	}
}

File: src/main/kotlin/com/gg/plugins/mongo/model/MongoCollection.kt
/*
 * Copyright (c) 2018 David Boissier.
 * Modifications Copyright (c) 2022 Geetesh Gupta.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.gg.plugins.mongo.model

import java.util.Objects

class MongoCollection(val name: String, val parentDatabase: MongoDatabase) : Comparable<MongoCollection> {
    override fun compareTo(other: MongoCollection): Int {
        return name.compareTo(other.name)
    }

    override fun hashCode(): Int {
        return Objects.hash(name, parentDatabase)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is MongoCollection) return false
        return name == other.name && parentDatabase == other.parentDatabase
    }

    override fun toString(): String {
        return name
        //		return "MongoCollection{" + "name='" + name + '\'' + '}';
    }
}